### PR TITLE
Added PermissionCollectionService, util and test

### PR DESCRIPTION
**Pull request by weskamm**
Added PermissionCollectionService, util and test

### CODE REVIEW

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PermissionCollectionService.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import de.terrestris.shogun2.dao.PermissionCollectionDao;
+import de.terrestris.shogun2.model.module.Module;
+import de.terrestris.shogun2.model.security.PermissionCollection;
+
+/**
+ * Service class for the {@link Module} model.
```
Comment (buehner) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/service/PermissionCollectionService.java]: This is wrong! :wink:

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/util/security/PermissionsUtil.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import de.terrestris.shogun2.model.security.Permission;
+import de.terrestris.shogun2.model.security.PermissionCollection;
+import de.terrestris.shogun2.service.PermissionCollectionService;
+
+/**
+*
+* @author Johannes Weskamm
+* @author terrestris GmbH & Co. KG
+*
+*/
```
Comment (marcjansen) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/util/security/PermissionsUtil.java]: nitpick: This docblock seems to be missing a space before the `*`

**Comment by buehner** [comment]
:+1: from me if the comment is fixed :wink:

File: src/shogun2-core/src/test/resources/META-INF/spring/mail-test-context.xml [hunk 3]
```diff
@@ -11,7 +11,7 @@
 
     <context:property-placeholder location="classpath*:META-INF/*.properties" />
 
-    <context:component-scan base-package="de.terrestris.shogun2.util" />
+    <context:component-scan base-package="de.terrestris.shogun2.util.mail" />
```
Comment (marcjansen) [inline src/shogun2-core/src/test/resources/META-INF/spring/mail-test-context.xml]: This isnÄt related to the title of the PR, is it?

Comment (buehner) [inline src/shogun2-core/src/test/resources/META-INF/spring/mail-test-context.xml]: No, but the mail test failed after the commit from before, so this fix should be included in this PR, shouldn't it?

Comment (marcjansen) [inline src/shogun2-core/src/test/resources/META-INF/spring/mail-test-context.xml]: OK for me. I was looking through the diff, not reading the commit message. Thanks for the clarification.

**Comment by weskamm** [comment]
Thanks, merging..

**Pull request merged**